### PR TITLE
feat: Added firebase logic

### PR DESCRIPTION
**Pull request by RahulSingh9131**
This branch contains logic for firebase implementation and several other components

### CODE REVIEW

File: pages/r/[clubId]/index.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import safeJsonStringify from "safe-json-stringify";
+import NotFound from '../../../components/Club/NotFound';
+import Header from '../../../components/Club/Header';
+import PageContent from '../../../components/Layout/PageContent';
+
+type clubPageProps = {
+    clubData: club,
+};
+
+const ClubPage:React.FC<clubPageProps> = ({clubData}) => {
```
Comment (rahulcs754) [inline pages/r/[clubId]/index.tsx:15]: what are you using here? this is new

**Review by rahulcs754 — APPROVED** [review]
The code looks good

File: pages/r/[clubId]/index.tsx (hunk 1 above)
Comment (RahulSingh9131) [reply pages/r/[clubId]/index.tsx:15]: added typeScript logic

**Review by shubhamkumar648 — APPROVED** [review]
- i really like you used typescript 
- well written code , good to merge

**Pull request merged**